src/repositories/Repository.ts

```diff
@@ -0,0 +1,102 @@
+import { IJsonApi, JsonAPIErrorResp, JsonAPIResp } from "../api/type";
+import { IRepository } from "./types";
+
+export class Repository implements IRepository {
+  private _api: IJsonApi;
+  constructor(api: IJsonApi) {
+    this._api = api;
+  }
+  private _getResponse<T>(response: JsonAPIResp<T>) {
+    return "data" in response
+      ? response.data
+      : "errors" in response
+      ? response
+      : undefined;
+  }
+  async registerUser(
+    userDetails: FormData | null
+  ): Promise<RegisterResponse | JsonAPIErrorResp | undefined> {
+    const responseData = await this._api.post<RegisterResponse>(
+      `/register/passenger`,
+      userDetails,
+      undefined,
+      false
+    );
+    return this._getResponse(responseData);
+  }
+
+  async registerDriver(
+    driverInfo: FormData | null
+  ): Promise<RegisterResponse | JsonAPIErrorResp | undefined> {
+    const responseData = await this._api.post<RegisterResponse>(
+      `/register/passenger`,
+      driverInfo,
+      undefined,
+      false
+    );
+    return this._getResponse(responseData);
+  }
+  async login(
+    loginInfo: FormData | null
+  ): Promise<JsonAPIErrorResp | loginResponse | undefined> {
+    const responseData = await this._api.post<loginResponse>(
+      `users/login`,
+      loginInfo,
+      undefined,
+      false
+    );
+    return this._getResponse(responseData);
+  }
+  async isPhoneExist(
+    phoneNumber: string
+  ): Promise<RegisterResponse | JsonAPIErrorResp | undefined> {
+    const responseData = await this._api.get<RegisterResponse>(
```

Comment:
Each of the following distinct methods should have a distinct response type. For eg. Response type for `isPhoneExist` could be `boolean`.